docs/FAQ.md

```diff
@@ -0,0 +1,19 @@
+# FAQ for s2n-tls
+
+### Why is the ClientHello version reported as TLS 1.2 but the handshake itself negotiated TLS 1.3?
+The TLS 1.3 spec uses the TLS 1.2 version in ClientHello and on record layers for compatibility with middleboxes and poor server implementations. There is a separate ClientHello extension which lists the actual versions supported.
+
+### Why is my TLS handshake failing on validating my peer’s certificate? I gave my s2n endpoint a root certificate that verifies the peer’s certificate.
+You may need to validate the peer’s hostname by implementing the s2n_verify_host_fn callback. The default behavior may need to be altered for your usecase.
+
+### Why is s2n hanging for so long before erroring?
+s2n-tls sleeps for 10-30 seconds after specific errors occur to prevent peers from learning about encrypted data. This technique is called blinding and it is utilized to prevent side-channel attacks.
+
+### Which security policy should I use if I want to make sure that it will never be altered?
+Our numbered security policies are guaranteed to never change. We will not alter or update them based on changing cryptography standards. However, our named security policies (like “default” or “default_tls13”) change based on new cryptography standards that come out. 
+
+### Does s2n-tls have an OpenSSL compatibility layer to make transitioning to s2n-tls easier? 
+s2n-tls does not provide compatibility with OpenSSL’s APIs. OpenSSL’s APIs are complex and creating a shim layer would take a lot of engineering effort. Currently we would rather focus on making our library as easy to use as possible and building useful features.
+
```

Comment:
Should we mention awslc?

Reply:
Like in what sense?

Reply:
awslc does mostly provide compatibility with OpenSSL's APIs, since it's forked from boringssl.